.travis.yml

```diff
@@ -38,6 +52,20 @@ matrix:
       compiler: clang
       env: [CONFIGURATION=Release, *gcc-warnings]
 
+  # Sporadic failure builds
+    - compiler: gcc
+      env: CONFIGURATION=Debug    CTEST_OPTS="--repeat-until-fail 25 --schedule-random"
+      install: *install_cmake35
+      cache: *cache_homedir
+    - compiler: gcc
+      env: CONFIGURATION=Release  CTEST_OPTS="--repeat-until-fail 25 --schedule-random"
+      install: *install_cmake35
+      cache: *cache_homedir
+
+  allow_failures:
+    - env: CONFIGURATION=Debug    CTEST_OPTS="--repeat-until-fail 25 --schedule-random"
+    - env: CONFIGURATION=Release  CTEST_OPTS="--repeat-until-fail 25 --schedule-random"
+
```

Comment:
I'm not sure to get what you're saying, are you asking if there's a way to avoid duplicating the full `env` entry?

If so, I don't think it's currently possible unfortunately. I originally added a `ALLOW_FAIL=yes` variable in both builds and put it as a condition for allowed failures, but it wasn't recognized. It seems you have to put the whole environment array as a condition.

What I could do, in fact, is cheat a little. I could specify a `compiler: gcc-failure` key in both jobs, then re-export `CC` as `gcc` rather than `gcc-failure`, and use the compiler key as the condition.

This is unfortunate hackery for what should be normal behaviour.